### PR TITLE
Base deployment: enabled metric endpoint for all controllers

### PR DESCRIPTION
**Pull request by verult**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/enable_provisioner_extra_metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# for external-provisioner
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: "--extra-create-metadata"
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/enable_provisioner_extra_metadata.yaml:4]: It's not enough to just enable the flag. We actually need to plumb the information that's passed in to PD creation.

Comment (verult) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/enable_provisioner_extra_metadata.yaml:4]: Right. Plumbing the information is a longer term task, and for the time being I wanted to first enable this flag and let it soak.

**Comment by verult** [comment]
The driver validates `CreateVolume()` params and rejects any call with unrecognized params, so the staging RC spec can only be updated with a PD CSI driver release that contains the fix for https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/387. Closing.

**Comment by verult** [comment]
Reopening and changing the PR to only include the metric change.

**Comment by verult** [comment]
Superseded by https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/517

**Pull request closed without merging**